app/ts/library/identifyTokens.ts

```diff
@@ -1,47 +1,143 @@
 	try {
-		if (isERC721.success === false || nftInterface.decodeFunctionResult('supportsInterface', isERC721.returnData)[0] === false) throw new Error('No ERC721 found at address')
-		if (owner.success === false || nftInterface.decodeFunctionResult('ownerOf', owner.returnData)[0] === ZeroAddress) throw new Error('No ERC721 found at address')
-	} catch {
-		// If any errors occur from deocoding returnData then it must not be ERC721
-		throw new Error('No ERC721 found at address')
-	}
-	if (!owner.success) throw new Error('Token ID does not exist')
-	return {
-		address,
-		id,
-		owner: nftInterface.decodeFunctionResult('ownerOf', owner.returnData)[0],
-		name: hasMetadata.success ? nftInterface.decodeFunctionResult('name', name.returnData)[0] : undefined,
-		tokenURI: hasMetadata.success ? nftInterface.decodeFunctionResult('tokenURI', tokenURI.returnData)[0] : undefined,
+
+		const [isERC721, hasMetadata, isERC1155, isERC1155Metadata, owner, name, symbol, decimals, totalSupply, tokenURI, erc1155Uri]: { success: boolean, returnData: BytesLike }[] = await multicall.tryAggregate.staticCall(false, calls)
+
+		if (isERC721.success && nftInterface.decodeFunctionResult('supportsInterface', isERC721.returnData)[0] === true) {
+			if (owner.success === false || nftInterface.decodeFunctionResult('ownerOf', owner.returnData)[0] === ZeroAddress) throw new Error('No ERC721 found at address')
+			return {
+				type: 'ERC721',
+				inputId: id,
+				address,
+				id,
+				owner: nftInterface.decodeFunctionResult('ownerOf', owner.returnData)[0],
+				name: hasMetadata.success ? nftInterface.decodeFunctionResult('name', name.returnData)[0] : undefined,
+				tokenURI: hasMetadata.success ? nftInterface.decodeFunctionResult('tokenURI', tokenURI.returnData)[0] : undefined,
+			}
+		}
+
+		if (isERC1155.success && nftInterface.decodeFunctionResult('supportsInterface', isERC1155.returnData)[0] === true) {
+			return {
+				type: 'ERC1155',
+				inputId: id,
+				address,
+				uri: isERC1155Metadata.success && nftInterface.decodeFunctionResult('supportsInterface', isERC1155Metadata.returnData)[0] === true && erc1155Uri.success ? erc1155Interface.decodeFunctionResult('uri', erc1155Uri.returnData)[0] : undefined,
+			}
+		}
+
+		if (name.success && decimals.success && symbol.success && totalSupply.success) {
+			return {
+				type: 'ERC20',
+				inputId: id,
+				address,
+				name: erc20Interface.decodeFunctionResult('name', name.returnData)[0],
+				symbol: erc20Interface.decodeFunctionResult('name', symbol.returnData)[0],
+				decimals: BigInt(erc20Interface.decodeFunctionResult('decimals', decimals.returnData)[0]),
+				totalSupply: erc20Interface.decodeFunctionResult('totalSupply', totalSupply.returnData)[0]
+			}
+		}
+
+	} catch (error) {
+		// For any reason decoding txing fails catch and return as unknown contract
```

Comment:
if we lose connection here, we map it incorrectly as contract too? These kind of catch alls are dangerous